### DISCUSSION TITLE
Local storage?

### DISCUSSION THREAD

**Discussion by NightHawkATL** [martadinata666/dockerized#2] [Q&A]
Can I get any assistance to get Reactive Resume to work with local storage? I want to keep this as non-cloud as possible and I have setup the local storage per the docker compose example but I can't get the photos to save and load. Everything else works great!

**Comment by martadinata666** [comment]
```
volumes:
  - /host/path:/home/debian/reactiveresume/server/dist/assets
```
and set your env
```
STORAGE_S3_ENABLED=false
```

**Reply by martadinata666** [reply]
And maybe share the compose, thanks

**Reply by NightHawkATL** [reply]
version: "3.8"
services:
  resume_app:
    image: martadinata666/reactive-resume:latest
    environment:
      - PUID=1000
      - PGID=1000
      - TZ=America/New_York
      - SECRET_KEY=[secret]
      - POSTGRES_HOST=resume_db
      - POSTGRES_PORT=5432
      - POSTGRES_USER=reactiveresume
      - POSTGRES_PASSWORD=[secret]
      - POSTGRES_DB=reactiveresume
      - PUBLIC_FLAG_DISABLE_SIGNUPS=true
      - JWT_SECRET=[secret]
      - JWT_EXPIRY_TIME=604800
      - STORAGE_S3_ENABLED=false
    ports:
      - 7680:80
      - 3000:3000
    networks:
      - reactiveresume
    volumes:
      - /portainer/Files/AppData/Config/react-resume/assets:/home/debian/reactiveresume/server/dist/assets #photo and export storage
    restart: unless-stopped 

  resume_db:
    image: postgres:13
    environment:
      - PUID=1000
      - PGID=1000
#      - POSTGRES_SSL_CERT=
      - POSTGRES_USER=reactiveresume
      - POSTGRES_PASSWORD=[secret]
      - POSTGRES_DB=reactiveresume
    restart: unless-stopped
    volumes:
      - /portainer/Files/AppData/Config/react-resume/db:/var/lib/postgresql/data
    ports:
      - 5432:5432
    networks:
      - reactiveresume 

networks:
  reactiveresume:
    name: reactiveresume

**Reply by martadinata666** [reply]
The permission of assets already 1000?

**Reply by NightHawkATL** [reply]
I tried with 1000, i tried with nothing and i tried with root. I domsee an error when tryimg tomadd the photo but it reads as no directory rather than an access error. I used portainer to spin up the container and set the variables the same as other contwiners for user permissions for the local host folder andi can see the folder without issue.
![20220713_223257](https://user-images.githubusercontent.com/8395658/178885687-e8db522a-ebf9-4b8d-8e18-632661dfcac9.png)

**Reply by martadinata666** [reply]
it run by 1000, likely there is old files with unmatched permission, you need chown everthing to 1000, 
```
chown -R  1000:1000 /portainer/Files/AppData/Config/react-resume/assets
```

**Reply by NightHawkATL** [reply]
Ok, I will try that in the morning and let you know. Thanks!

**Reply by martadinata666** [reply]
In my system the assets only contain `uploads` folder, you can also try just mount the `uploads` folder
```
  volumes:
    - /portainer/Files/AppData/Config/react-resume/uploads:/home/debian/reactiveresume/server/dist/assets/uploads
```

**Reply by NightHawkATL** [reply]
> it run by 1000, likely there is old files with unmatched permission, you need chown everthing to 1000,
> 
> ```
> chown -R  1000:1000 /portainer/Files/AppData/Config/react-resume/assets
> ```

This ended up fixing it for me. Sorry for the late reply. Feel free to close.
